examples/speech_text_joint_to_text/models/s2t_dualinputtransformer.py

```diff
@@ -850,6 +850,7 @@ def build_decoder(cls, args, task):
             "adaptive_softmax_cutoff": args.adaptive_softmax_cutoff,
             "tie_adaptive_weights": args.tie_adaptive_weights,
             "no_token_positional_embeddings": args.no_token_positional_embeddings,
+            "encoder": {"embed_dim":args.encoder_embed_dim}
```

Comment:
What's the context of this change? Are you using different embedding for encoder and decoder?

Reply:
No, I used the recommended command line from "joint text" example to run this (I pasted the exact command line into Bug #3939), but it broke here due to a dimension mismatch.

I think the base class changed sometime this year, breaking the example, but I didn't isolate the root cause. Should I?

Reply:
Thanks for reporting this. Yes, it is due to the base class change and it will assign default embedding dim as 512 for both encoder and decoder. The attention between encoder and decoder will load the default value if no encoder embed_dim is passed.